modules/gapi/test/common/gapi_tests_common.hpp

```diff
@@ -155,12 +155,58 @@ class TestFunctional
         }
     }
 
+    void initMatsFromImages(int channels, const std::string& fileNamePattern, int format)
+    {
+        initTestDataPath();
+        switch (channels)
+        {
+            case 1:
+                in_mat1 = cv::imread(perf::TestBase::getDataPath(cv::format(fileNamePattern.c_str(),
+                                                                            format)),
+                                     cv::IMREAD_GRAYSCALE);
+                in_mat2 = cv::imread(perf::TestBase::getDataPath(cv::format(fileNamePattern.c_str(),
+                                                                            format + 1)),
+                                     cv::IMREAD_GRAYSCALE);
+                break;
+            case 3:
+                in_mat1 = cv::imread(perf::TestBase::getDataPath(cv::format(fileNamePattern.c_str(),
+                                                                            format)));
+                in_mat2 = cv::imread(perf::TestBase::getDataPath(cv::format(fileNamePattern.c_str(),
+                                                                            format + 1)));
+                break;
+            case 4:
+                cvtColor(cv::imread(perf::TestBase::getDataPath(cv::format(fileNamePattern.c_str(),
+                                                                           format))),
+                         in_mat1, cv::COLOR_BGR2BGRA);
+                cvtColor(cv::imread(perf::TestBase::getDataPath(cv::format(fileNamePattern.c_str(),
+                                                                           format + 1))),
+                         in_mat2, cv::COLOR_BGR2BGRA);
+                break;
+            default:
+                FAIL() << "Unexpected number of channels: " << channels;
+        }
+    }
+
     // empty function intended to show that nothing is to be initialized via TestFunctional methods
     void initNothing(int, cv::Size, int, bool = true) {}
-};
 
-template<class T>
-class TestParams: public TestFunctional, public TestWithParam<T>{};
+    void initTestDataPath()
+    {
+#ifndef WINRT
+        static bool initialized = false;
+        if (!initialized)
+        {
+            // Since G-API has no own test data (yet), it is taken from the common space
+            const char* testDataPath = getenv("OPENCV_TEST_DATA_PATH");
+            GAPI_Assert(testDataPath != nullptr &&
+            "OPENCV_TEST_DATA_PATH environment variable is either not set or set incorrectly.");
+
+            cvtest::addDataSearchPath(testDataPath);
+            initialized = true;
+        }
+#endif // WINRT
```

Comment:
So now we need people to have `opencv_extra` repo available by default. Not sure if this is a good idea. 99% of our tests don't need that.

Reply:
The function isn't called by default, only by user. Should I leave it as is or write a comment to use it wisely not to create problems?

Reply:
Better to move this code out of here

Reply:
Will it be better if I move it to the unnamed namespace in this file (str. 22)? I need this function not only in tests of `video::` kernels, but also in tests `imgproc::goodFeatures`, so I want to preserve the function in this file

Reply:
anonymous namespace in a header doesn't really solve _this_ problem (but for sure solves a number of others)